app/src/main/java/com/example/bookwormadventuresdeluxe2/MyBooksDetailViewFragment.java

```diff
@@ -123,24 +123,37 @@ public void onEditClick(View v)
         startActivityForResult(intent, AddOrEditBooksActivity.EDIT_BOOK);
     }
 
-    // Called by AddOrEditBooksActivity when user presses save on editing screen
+    /* Called by AddOrEditBooksActivity when user presses save or delete on edit screen */
     @Override
     public void onActivityResult(int requestCode, int resultCode, Intent data)
     {
         super.onActivityResult(requestCode, resultCode, data);
         getActivity();
-        if (requestCode == AddOrEditBooksActivity.EDIT_BOOK && resultCode == Activity.RESULT_OK)
+        if (requestCode == AddOrEditBooksActivity.EDIT_BOOK)
         {
-            // Get the book that was edited and its new values
-            this.selectedBook = (Book) data.getSerializableExtra("EditedBook");
-            updateView(this.selectedBook);
-
-            // Update the book in firebase
-            FirebaseFirestore rootRef = FirebaseFirestore.getInstance();
-            rootRef.collection("Books").document(this.selectedBookId).update("title", this.selectedBook.getTitle());
-            rootRef.collection("Books").document(this.selectedBookId).update("author", this.selectedBook.getAuthor());
-            rootRef.collection("Books").document(this.selectedBookId).update("description", this.selectedBook.getDescription());
-            rootRef.collection("Books").document(this.selectedBookId).update("isbn", this.selectedBook.getIsbn());
+            /* Save was pressed */
+            if (resultCode == Activity.RESULT_OK)
+            {
+                // Get the book that was edited and its new values
+                this.selectedBook = (Book) data.getSerializableExtra("EditedBook");
+                updateView(this.selectedBook);
+
+                /* Update the book in firebase */
+                DocumentReference bookDocument = FirebaseFirestore
+                        .getInstance()
+                        .collection(getString(R.string.books_collection))
+                        .document(this.selectedBookId);
+
+                bookDocument.update("title", this.selectedBook.getTitle());
+                bookDocument.update("author", this.selectedBook.getAuthor());
+                bookDocument.update("description", this.selectedBook.getDescription());
+                bookDocument.update("isbn", this.selectedBook.getIsbn());
+            }
+            else if (resultCode == Activity.RESULT_CANCELED) /* Delete was pressed */

```

Comment:
Related to this comment:
Activity.RESULT_OK is just a public static int inside the Activity class.
Maybe we should create a public static int BOOK_EDITED and BOOK_DELETED inside AddOrEditBookActivity for these.
It would more clearly reflect what is going on